backend/test_observer/external_apis/c3/c3.py

```diff
@@ -0,0 +1,74 @@
+import logging
+import os
+from collections.abc import Iterable
+
+import requests
+from requests import Request, Response
+
+from .models import Report, SubmissionStatus
+
+logger = logging.getLogger("test-observer-backend")
+
+
+class C3Api:
+    def __init__(self):
+        self._client_id = os.environ.get("C3_CLIENT_ID", "")
+        self._client_secret = os.environ.get("C3_CLIENT_SECRET", "")
+        self._bearer_token = ""
+
+    def get_submissions_statuses(
+        self, ids: Iterable[int]
+    ) -> dict[int, SubmissionStatus]:
+        str_ids = [str(status_id) for status_id in ids]
+        response = self._authenticate_and_send(
+            Request(
+                method="GET",
+                url="https://certification.canonical.com/api/v2/submissions/status/",
+                params={"id__in": ",".join(str_ids), "limit": len(str_ids)},
```

Comment:
right now limit doesn't work. should this PR depend on this change (to add `limit` param for v2 API) in c3?